### DISCUSSION TITLE
Neopixel simulation

### DISCUSSION THREAD

**Discussion by ScobbyDoo** [micropython/micropython#15681] [Using MicroPython]
https://github.com/dgatf/pio-ws2812-rp2040/issues/1

I've created issue here , 

I've tried with 2 logic, 
1st is using pwm width calculation 
2nd get timings of signals with reset pulse

I'm little bit confused for setting pio in micropython

When reset pulse detects , >50us
Need to send all stored data in serially & on other side I'll receive that data & ahows it on pygame canvas with dynamically decoded led counts between reset pulses 

Any suggestions..
Thanks

**Comment by GitHubsSilverBullet** [comment]
> When reset pulse detects , >50us

Forget that.

You're at the start of a sequence.
Send data for all LEDs connected sequentially (ie. 24bits per LED).
Once you've sent enough data, just make a >=50µs pause which automagically resets the LEDs internal logic.
Then you're once again at the start of the next sequence.

Data for one LED takes 1.25µs. If you have a 1000 LED chain, it sums up to 1.250ms+0.050ms.
That would allow for a max rate of about 769 FPS. (1/1.3e-3)
